### PR TITLE
Dt use patch

### PR DESCRIPTION
**Pull request by kristapratico**
typespec client.tsp changes: https://github.com/Azure/azure-rest-api-specs/pull/30416

### CODE REVIEW

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_operations/_operations.py [hunk 1]
```diff
@@ -260,7 +260,7 @@ def build_document_translation_get_documents_status_request(  # pylint: disable=
 
 
 def build_document_translation_get_supported_formats_request(  # pylint: disable=name-too-long
-    *, type: Optional[Union[str, _enums.FileFormatType]] = None, **kwargs: Any
+    *, type: Optional[Union[str, _models.FileFormatType]] = None, **kwargs: Any
```
Comment (catalinaperalta) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_operations/_operations.py:263]: This looks like an error in the generated code? I see FileFormatType in _enums.py

Comment (catalinaperalta) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_operations/_operations.py:263]: Ah nevermind I just saw the import...resolving

**Review by hamshavathimunibyraiah — APPROVED** [review]

**Review by catalinaperalta — COMMENTED** [review]
@kristapratico can you open a draft pr for your typespec changes as well? I had a few comments about the changes.

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_operations/_operations.py [hunk 2]
```diff
@@ -317,9 +319,9 @@ def build_single_document_translation_document_translate_request(  # pylint: dis
 
 class DocumentTranslationClientOperationsMixin(DocumentTranslationClientMixinABC):
 
-    def _start_translation_initial(  # pylint: disable=inconsistent-return-statements
+    def __begin_start_translation_initial(
         self, body: Union[_models.StartTranslationDetails, JSON, IO[bytes]], **kwargs: Any
-    ) -> None:
+    ) -> Iterator[bytes]:
```
[the remark this answers is not in the document]
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_operations/_operations.py:324]: Seems to be from a generator update.

Draft PR for tsp changes: https://github.com/Azure/azure-rest-api-specs/pull/30416

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_operations/_patch.py [hunk 3]
```diff
@@ -46,10 +57,228 @@
...
 
+ResponseType = Union[HttpResponse, AsyncHttpResponse]
+PipelineResponseType = PipelineResponse[HttpRequest, ResponseType]
+PollingReturnType_co = TypeVar("PollingReturnType_co", covariant=True)
+
+_FINISHED = frozenset(["succeeded", "cancelled", "cancelling", "failed"])
+_FAILED = frozenset(["validationfailed"])
+
+
+def convert_status(status, ll=False):
```
Comment (catalinaperalta) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_operations/_patch.py:68]: Nit: was this method public in the old sdk? Wondering if it needs an `_` in the name?

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py [hunk 4]
```diff
@@ -1036,16 +275,110 @@ def begin_translation(
 
     @overload
     def begin_translation(
-        self, inputs: List[DocumentTranslationInput], **kwargs: Any
+        self, inputs: StartTranslationDetails, **kwargs: Any
```
Comment (catalinaperalta) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py:278]: Was the previous type incorrect here?

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py [hunk 5]
```diff
@@ -1036,16 +275,110 @@ def begin_translation(
...
 
-        1) To perform translation on documents from a single source container to a single target container, pass the
-        `source_url`, `target_url`, and `target_language` parameters including any optional keyword arguments.
+        For supported languages and document formats, see the service documentation:
+        https://docs.microsoft.com/azure/cognitive-services/translator/document-translation/overview
+
+        :param inputs: A StartTranslationDetails including translation inputs. Each individual input has a single
+            source URL to documents and can contain multiple TranslationTargets (one for each language)
+            for the destination to write translated documents.
+        :type inputs: List[~azure.ai.translation.document.models.StartTranslationDetails]
```
Comment (catalinaperalta) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py:289]: This says `inputs` is a list whereas the param type doesnt have a list

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py [hunk 6]
```diff
@@ -1036,16 +275,110 @@ def begin_translation(
...
+            object to return a pageable of DocumentStatus. A DocumentStatus will be
+            returned for each translation on a document.
+        :rtype: DocumentTranslationLROPoller[~azure.core.paging.ItemPaged[DocumentStatus]]
+        :raises ~azure.core.exceptions.HttpResponseError:
+    """
+
+    @overload
+    def begin_translation(
+        self, inputs: List[DocumentTranslationInput], **kwargs: Any
+    ) -> DocumentTranslationLROPoller[ItemPaged[DocumentStatus]]:
```
Comment (catalinaperalta) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py:379]: Is this still a valid overload?

Comment (catalinaperalta) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py:379]: ok looks like it is based on the base method.

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_operations/_patch.py [hunk 7]
```diff
@@ -46,10 +57,228 @@
...
+        http_response_error = HttpResponseError(message="({}): {}".format(error["code"], error["message"]))
+        http_response_error.error = ODataV4Format(error)  # set error.code
+        raise http_response_error
+
+
 class DocumentTranslationClientOperationsMixin(GeneratedDocumentTranslationClientOperationsMixin):
 
     @distributed_trace
-    def begin_start_translation(  # type: ignore[override]
+    def _begin_start_translation(  # type: ignore[override]
```
Comment (catalinaperalta) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_operations/_patch.py:281]: Are we still patching this method on the mixin due to the custom poller?

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py [hunk 8]
```diff
@@ -1162,7 +515,7 @@ def cancel_translation(self, translation_id: str, **kwargs: Any) -> None:
         :raises ~azure.core.exceptions.HttpResponseError or ~azure.core.exceptions.ResourceNotFoundError:
         """
 
-        self._client.cancel_translation(translation_id, **kwargs)
+        super().cancel_translation(translation_id, **kwargs)
 
     @distributed_trace
     def list_translation_statuses(
```
Comment (catalinaperalta) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py:521]: Do we still need to patch this operation?

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py [hunk 9]
```diff
@@ -1317,8 +649,11 @@ def _convert_from_generated_model(generated_model):
             ),
         )
 
+    # pylint: disable=arguments-renamed
     @distributed_trace
-    def get_document_status(self, translation_id: str, document_id: str, **kwargs: Any) -> DocumentStatus:
+    def get_document_status(  # type: ignore
```
Comment (catalinaperalta) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py:654]: Does the generated client expose this method? And if so, do we still need to patch it here?

File: sdk/translation/azure-ai-translation-document/tests/test_translation_async.py [hunk 10]
```diff
@@ -463,8 +463,8 @@ def callback(request):
             assert input.source.filter.prefix == ""
             assert input.source.filter.suffix == ".txt"
             assert input.storage_type == "File"
-            assert input.targets[0].category == "fake"
-            assert input.targets[0].glossaries[0].format == "txt"
+            assert input.targets[0].category_id == "fake"
```
Comment (annatisch) [inline sdk/translation/azure-ai-translation-document/tests/test_translation_async.py:466]: These two attributes here are breaking API changes?

Comment (annatisch) [reply sdk/translation/azure-ai-translation-document/tests/test_translation_async.py:466]: Actually it seems in the APIview diff that these two properties were always called `category_id` and `file_format`?

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/aio/_patch.py [hunk 11]
```diff
@@ -410,8 +380,9 @@ def deserialization_callback(raw_response, _, headers):  # pylint: disable=unuse
             ),
         )
 
+    # pylint: disable=arguments-renamed
     @distributed_trace_async
-    async def get_translation_status(self, translation_id: str, **kwargs: Any) -> TranslationStatus:
+    async def get_translation_status(self, translation_id: str, **kwargs: Any) -> TranslationStatus:  # type: ignore
```
[the remark this answers is not in the document]
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/aio/_patch.py]: Your guess is correct - I will add the error type for all instances.

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_enums.py [hunk 12]
```diff
@@ -30,9 +30,9 @@ class Status(str, Enum, metaclass=CaseInsensitiveEnumMeta):
     """Succeeded"""
     FAILED = "Failed"
     """Failed"""
-    CANCELLED = "Cancelled"
+    CANCELED = "Cancelled"
```
[the remark this answers is not in the document]
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_enums.py:33]: We didn't expose this in the GA version, nor the last beta. It is public by default for the current version that leverages the codegen models.

The reason I renamed the statuses here was to have consistency throughout the library. In the GA version, we decided we were against "cancelled" with a double `l` (don't ask me why, I don't remember 😆) and have a conversion function that changes any double `l` cancel to a single `l` in the response.

File: sdk/translation/azure-ai-translation-document/tests/test_translation_async.py (hunk 10 above)
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/tests/test_translation_async.py:466]: Sorry it's a bit confusing - we renamed a lot of things in the GA version of this library. `StartTranslationDetails` (the type we're accessing here) was not exposed in the GA library. We didn't expose this in the beta release following the GA either, but it was used here in a test to verify deserialization.

In this PR, we make StartTranslationDetails public, as another input option to begin_translation. To be consistent with the renaming we did in the GA library, I've renamed these two properties to match.

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py (hunk 9 above)
Comment (catalinaperalta) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py]: Ah I see...yes that is an unfortunate limitation we've been dealing with, but since I think it's more noise to patch in a client library instead of just using the `@clientName` decorator, I'd say just use the decorator over the typespec property inline in the code. And make sure to specify Python if it's the only language that renamed it.

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/__init__.py [hunk 13]
```diff
@@ -6,13 +6,49 @@
...
 # --------------------------------------------------------------------------
 
+from ._models import BatchRequest
+from ._models import DocumentFilter
+from ._patch import DocumentStatus
 from ._models import DocumentTranslateContent
-from ._patch import __all__ as _patch_all
-from ._patch import *  # pylint: disable=unused-wildcard-import
+from ._models import DocumentTranslationFileFormat
+from ._patch import TranslationGlossary
```
Comment (annatisch) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/__init__.py:14]: Has this been manually edited?

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py [hunk 14]
```diff
@@ -7,9 +7,472 @@
...
+    """Category / custom system for translation request."""
+    language: str
+    """Target Language. Required."""
+    glossaries: Optional[List["TranslationGlossary"]]
+    """List of Glossary."""
+    storage_source: Optional[Union[str, "_models.StorageSource"]]
+    """Storage Source. \"AzureBlob\""""
+
+    @overload
+    def __init__(
```
Comment (annatisch) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py:173]: This overload shouldn't be needed.

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py [hunk 15]
```diff
@@ -7,9 +7,472 @@
...
-__all__: List[str] = []  # Add all objects you want publicly available to users at this package level
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from .. import models as _models
+
+
+def convert_status(status, ll=False):
+    if ll is False:
+        if status == "Cancelled":
```
Comment (annatisch) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py:31]: We could probably make these checks case insensitive:
`if status.lower() == 'cancelled'`

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py [hunk 16]
```diff
@@ -7,9 +7,472 @@
...
+     language pair is not present in the glossary, it will not be applied. Required."""
+    file_format: str
+    """Format. Required."""
+    format_version: Optional[str]
+    """Optional Version.  If not specified, default is used."""
+    storage_source: Optional[Union[str, "_models.StorageSource"]]
+    """Storage Source. \"AzureBlob\""""
+
+    @overload
+    def __init__(
```
Comment (annatisch) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py:245]: I think this overload isn't needed.

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_models.py [hunk 17]
```diff
@@ -360,152 +508,207 @@ class StatusSummary(_model_base.Model):
     """Number of in progress. Required."""
     not_yet_started: int = rest_field(name="notYetStarted")
     """Count of not yet started. Required."""
-    cancelled: int = rest_field()
+    canceled: int = rest_field(name="cancelled")
```
Comment (annatisch) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_models.py:511]: We don't rename body parameters. Why was this changed?

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/aio/_patch.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/aio/_patch.py]: Krista has updated this to : # type: ignore[override]

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/__init__.py (hunk 13 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/__init__.py:14]: I tried re-generating the SDK and there were no changes in this file, so NO, it was not manually edited

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py (hunk 15 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py:31]: Updated

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py (hunk 14 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py]: Deleted

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py (hunk 15 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py]: Updated

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py (hunk 16 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_patch.py:234]: Deleted

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py (hunk 9 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py]: Added the decorator in routes.tsp on the id parameter and regenerated the SDK and cleaned up the patching for those operations.

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_models.py (hunk 17 above)
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/models/_models.py:511]: This is response only. Changed for back compatibility with GA version.

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py [hunk 18]
```diff
@@ -1130,40 +484,6 @@ def deserialization_callback(raw_response, _, headers):  # pylint: disable=unuse
...
-        If possible, all pending documents will be canceled.
-
-        :param str translation_id: The translation operation ID.
-        :return: None
-        :rtype: None
-        :raises ~azure.core.exceptions.HttpResponseError or ~azure.core.exceptions.ResourceNotFoundError:
-        """
-
-        self._client.cancel_translation(translation_id, **kwargs)
-
```
Comment (kristapratico) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py:1166]: I think we need to keep cancel_translation in the _patch. The return type has changed when the generated code is used. 

GA --> returns None
this version --> returns TranslationStatus

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/aio/_operations/_patch.py [hunk 19]
```diff
@@ -48,11 +54,144 @@
...
+
+    async def _poll(self) -> None:
+        """Poll status of operation so long as operation is incomplete and
+        we have an endpoint to query.
+
+        :raises: OperationFailed if operation status 'Failed' or 'Canceled'.
+        :raises: BadStatus if response status invalid.
+        :raises: BadResponse if response invalid.
+        """
+        while not self.finished():
```
Comment (kristapratico) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/aio/_operations/_patch.py:176]: Just noticing this and it looks like a bug, please check the other patched polling class too.

```suggestion
        if not self.finished():
```

Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/aio/_operations/_patch.py]: Updated

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py (hunk 18 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py]: Added back
